components/Session.php

```diff
@@ -88,7 +88,10 @@ public function init()
     public function onRun()
     {
         if (!$this->checkUserSecurity()) {
-            $redirectUrl = $this->controller->pageUrl($this->property('redirect'));
+            if ($redirect = $this->property('redirect') == '') {
```

Comment:
Did you test this actually running?

Reply:
just simplify the if to `if (empty($this->property('redirect'))` and change the pageUrl call to `$this->controller->pageUrl($this->property('redirect'))`. Assigning variables in conditional checking statements is messy and not really necessary for this case.